src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -4153,6 +4153,208 @@ .maxstack  2
 ");
         }
 
+        [Fact]
+        public void SwitchExpressionAsExceptionFilter()
+        {
+            var source = @"
+using System;
+class C
+{
+    const string K1 = ""frog"";
+    const string K2 = ""toad"";
+    public static void M(string msg)
+    {
+        try
+        {
+            T(msg);
+        }
+        catch (Exception e) when (e.Message switch
```

Comment:
What are the various expression types which could cause the spiller to come into play. The ones I could recall are:

1. stackalloc as a parameter (think that invokoes the spiller)
1. await 
1. patterns

The first two are illegal in a filter block. Wanted to make sure there wasn't another expression form that uses spills that we should consider testing here.

Reply:
That's it, today. I've added a note to the test plan in case there are others in the future.

---
In reply to: [456699320](https://github.com/dotnet/roslyn/pull/46054#discussion_r456699320) [](ancestors = 456699320)